client/blockchain-service/src/utils.rs

```diff
@@ -107,6 +107,45 @@ impl BlockchainService {
         }
     }
 
+    pub(crate) fn register_best_block_and_check_reorg<Block>(
```

Comment:
is is it possible to write unit tests for this?
I think it would be worth it

Reply:
Although I could write a unit test _specifically_ for this function, we don't have unit tests for Blockchain Service functions in general. Anyway, my goal is to test this alongside the full re-org behaviour in the integration tests. I include some tests for this already in this PR, but more will come as this feature progresses.